HashTagTokenizer.java

```diff
@@ -7,33 +7,45 @@ public static void main(String[] args) {
 		String hashTag = args[0];
 		String []dictionary = readDictionary("dictionary.txt");
 		breakHashTag(hashTag, dictionary);
+
 	}
 
 	public static String[] readDictionary(String fileName) {
 		String[] dictionary = new String[3000];
-
+		int i = 0;
 		In in = new In(fileName);
 
-		// Your code here
+		while (!in.isEmpty()) {
```

Comment:
Protect your code from unexpected behaviors.
```suggestion
		while (!in.isEmpty() && I < dictionary.length) {
```